core/src/hir/attrs.rs

```diff
@@ -0,0 +1,157 @@
+//! #[diplomat::attr] and other attributes
+
+use crate::ast;
+use crate::ast::attrs::DiplomatAttrCfg;
+use crate::hir::LoweringError;
+
+use syn::Meta;
+
+#[non_exhaustive]
+#[derive(Clone, Default, Debug)]
+pub struct Attrs {
+    pub disable: bool,
+    // more to be added: rename, namespace, etc
+}
+
+/// Where the attribute was found. Some attributes are only allowed in some contexts
+/// (e.g. namespaces cannot be specified on methods)
+#[non_exhaustive] // might add module attrs in the future
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum AttributeContext {
+    Struct { out: bool },
+    Enum,
+    Opaque,
+    Method,
+}
+
+impl Attrs {
+    pub fn from_ast(
+        ast: &ast::Attrs,
+        validator: &(impl AttributeValidator + ?Sized),
+        _context: AttributeContext,
+        errors: &mut Vec<LoweringError>,
```

Comment:
Question: are you going to change this after merge to support the new ErrorContext?

Reply:
ErrorStore is for tool code, so no. We may need an equivalent ErrorContext type for AST stuff though to improve error provenance. Out of scope here